src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,81 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  id,
+  title,
+  completed,
+  userId,
+  onClose,
+}: Props) => {
+  const [userData, setUserData] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      setIsLoading(true);
+
+      try {
+        const user = await getUser(userId);
+
+        setUserData(user);
+      } catch (err) {
+        throw new Error();
```

Comment:
Throwing a generic error here is not informative. Consider handling the error more gracefully, such as logging the error or displaying a user-friendly message to the user.